src/main/java/org/spectra/cluster/tools/SpectraClusterTool.java

```diff
@@ -0,0 +1,219 @@
+            if (commandLine.hasOption(CliOptions.OPTIONS.ROUNDS.getValue()))
+                rounds = Integer.parseInt(commandLine.getOptionValue(CliOptions.OPTIONS.ROUNDS.getValue()));
+
+            // START THRESHOLD
+            float startThreshold = defaultParameters.getThresholdStart();
+            if (commandLine.hasOption(CliOptions.OPTIONS.START_THRESHOLD.getValue()))
+                startThreshold = Float.parseFloat(commandLine.getOptionValue(CliOptions.OPTIONS.START_THRESHOLD.getValue()));
+
+            // END THRESHOLD
+            float endThreshold = defaultParameters.getThresholdEnd();
+            if (commandLine.hasOption(CliOptions.OPTIONS.END_THRESHOLD.getValue()))
+                endThreshold = Float.parseFloat(commandLine.getOptionValue(CliOptions.OPTIONS.END_THRESHOLD.getValue()));
+
+            // PRECURSOR TOLERANCE
+            double precursorTolerance = defaultParameters.getPrecursorIonTolerance();
+            if (commandLine.hasOption(CliOptions.OPTIONS.PRECURSOR_TOLERANCE.getValue())) {
+                precursorTolerance = Float.parseFloat(commandLine.getOptionValue(CliOptions.OPTIONS.PRECURSOR_TOLERANCE.getValue()));
+            }
+
+            // PRECURSOR TOLERANCE
+            double fragmentTolerance = defaultParameters.getFragmentIonTolerance();
+            if (commandLine.hasOption(CliOptions.OPTIONS.FRAGMENT_TOLERANCE.getValue())) {
+                fragmentTolerance = Float.parseFloat(commandLine.getOptionValue(CliOptions.OPTIONS.FRAGMENT_TOLERANCE.getValue()));
+            }
+
+            int nInitiallySharedPeaks = defaultParameters.getNInitiallySharedPeaks();
+
+            /** Perform clustering **/
+            IRawSpectrumFunction loadingFilter = new RemoveImpossiblyHighPeaksFunction()
+                    .specAndThen(new RemovePrecursorPeaksFunction(fragmentTolerance))
+                    .specAndThen(new RawPeaksWrapperFunction(new KeepNHighestRawPeaks(defaultParameters.getNumberHigherPeaks())));
+
+            IPropertyStorage localStorage = PropertyStorageFactory.buildDynamicPropertyStorage();
+            File[] inputFiles = Arrays.stream(peakFiles)
+                    .map(File::new)
+                    .toArray(File[]::new);
+
+            MzSpectraReader reader = new MzSpectraReader( new TideBinner(), new MaxPeakNormalizer(),
+                    new BasicIntegerNormalizer(), new HighestPeakPerBinFunction(), loadingFilter, inputFiles);
+            Iterator<IBinarySpectrum> iterator = reader.readBinarySpectraIterator(localStorage);
+            List<IBinarySpectrum> spectra = new ArrayList<>(1_000);
+
+            while (iterator.hasNext()) {
+                spectra.add(iterator.next());
+            }
+
+            // sort according to m/z
+            spectra.sort(Comparator.comparingInt(IBinarySpectrum::getPrecursorMz));
+
+            // cluster everything
+            float[] thresholds = {0.99f, 0.98f, 0.95f, 0.995f};
```

Comment:
In the final version, this needs to be changed to the user defined final threshold